plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergBucketFunction.java

```diff
@@ -87,6 +127,12 @@ public int getBucket(Page page, int position)
         for (int i = 0; i < partitionColumns.size(); i++) {
             PartitionColumn partitionColumn = partitionColumns.get(i);
             Block block = page.getBlock(partitionColumn.getSourceChannel());
+            if (!partitionColumn.getPath().isEmpty()) {
+                for (int index : partitionColumn.getPath()) {
+                    ColumnarRow columnarRow = toColumnarRow(block);
+                    block = columnarRow.getField(index);
```

Comment:
Would `block = block.getChildren().get(index)` work ?

Reply:
It is suggested by @alexjo2144 to use `ColumnarRow` https://github.com/trinodb/trino/pull/15712#discussion_r1088195967